scheduler/src/cook/mesos/api.clj

```diff
@@ -1679,6 +1679,65 @@
                      (get-in ctx [:request :body-params :reason])
                      (reduce into [] (::limits ctx))))}))
 
+(def UserUsageParams
+  "User usage endpoint query string parameters"
+  (assoc UserParam
+         (s/optional-key :group_breakdown) s/Bool))
+
+(def UsageInfo
+  "Schema for a dictionary of job resource usage info."
+  {:cpus s/Num, :mem s/Num, :gpus s/Num, :jobs s/Num})
+
+(def UsageGroupInfo
+  "Helper-schema for :grouped jobs' group info in UserUsageResponse."
+  {:uuid s/Uuid, :name s/Str, :running_jobs [s/Uuid]})
+
+(def JobsUsageResponse
+  "Helper-schema for :ungrouped jobs in UserUsageResponse."
+  {:running_jobs [s/Uuid], :usage UsageInfo})
+
+(def UserUsageResponse
+  "Schema for a usage response."
+  {:total_usage UsageInfo
+   (s/optional-key :grouped) [{:group UsageGroupInfo, :usage UsageInfo}]
+   (s/optional-key :ungrouped) JobsUsageResponse})
+
+(def zero-usage
+  "Resource usage map 'zero' value"
+  (util/total-resources-of-jobs nil))
+
+(defn get-user-usage
+  "Query a user's current resource usage based on running jobs."
+  [db ctx]
+  (let [user (get-in ctx [:request :query-params :user])
+        running-jobs (mapv :job/_instance (util/get-user-running-task-ents db user))
```

Comment:
Does it make sense to add support for waiting instead of running jobs?

Reply:
Do you want to open a ticket for that? I can't think of a compelling use case, but you know our users better than I do.

Reply:
I can't either, so let's wait and see if anyone asks for it